### ISSUE TITLE
java.lang.NoSuchMethodError: 'void javax.ws.rs.core.MultivaluedMap.addAll(java.lang.Object, java.lang.Object[])' when used in Jenkins

### ISSUE THREAD

**Issue by kwin** [adobe/aio-lib-java-cloudmanager#20]
I use the client in a Jenkins Shared Library via 
```
@Grab(group='io.adobe.cloudmanager', module='aio-lib-cloudmanager', version='0.4.0')
```

When trying to get the token I run into the following exception

```
java.lang.NoSuchMethodError: 'void javax.ws.rs.core.MultivaluedMap.addAll(java.lang.Object, java.lang.Object[])'
	at org.glassfish.jersey.client.ClientRequest.accept(ClientRequest.java:322)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.accept(JerseyInvocation.java:224)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.accept(JerseyInvocation.java:147)
	at io.adobe.cloudmanager.jwt.swagger.invoker.ApiClient.invokeAPI(ApiClient.java:696)
	at io.adobe.cloudmanager.jwt.swagger.api.JwtApi.authenticate(JwtApi.java:109)
	at io.adobe.cloudmanager.impl.IdentityManagementApiImpl.authenticate(IdentityManagementApiImpl.java:73)
	at io.adobe.cloudmanager.impl.IdentityManagementApiImpl.authenticate(IdentityManagementApiImpl.java:67)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:47)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at customGroovyscript.....
```

The reason for that is that the Java library requires JAX-RS 2.0.0 while Jenkins still seems to ship with JAX-RS 1.1.0 (or older): https://stackoverflow.com/a/44070033.

**Comment by kwin** [comment]
Was there a particular reason why `jackson2` was used as library (https://github.com/adobe/aio-lib-java-cloudmanager/blob/8ad4eab0363fdc02a95681a729878d48bc53ab8e/pom.xml#L131). To me it seems that the default http client + json parser `okhttp-gson` could be less conflicting, as it doesn't require JAX RS 2.0.

**Comment by kwin** [comment]
The classpath in Jenkins seems to be a mess. A similar issue was reported  in https://issues.jenkins.io/browse/JENKINS-44378. I am still unsure which plugin include JAX-RS 1.x or whether this is provided by the Jenkins Core.

**Comment by bstopp** [comment]
Jersey2 was used as I was most familiar with it.

This issue seems to be specific to using the library in the context of Jenkins. While it is a/the primary use case for this, it is an issue related to that context, and not necessarily something that this project needs to resolve.

As a side note, from the Jenkins documentation:

> While possible, accessing third-party libraries using @Grab from trusted libraries has various issues and is not recommended.

[Link](https://www.jenkins.io/doc/book/pipeline/shared-libraries/#using-third-party-libraries)

**Comment by kwin** [comment]
When you recommend using an external binary one could use the aio-cli in the first place. I would still say that executing this library inside a Jenkins Shared Library should be supported (deploying an external tool requires some effort). Let me try to come up with a fix which relies on shading and relocation

**Comment by kwin** [comment]
Just for the reference: The Groovy Shared Library uses a class loader which merges all plugin classloaders. This is a parent class loader of the Grapes class loader and therefore used preferably. In my case it was the [JIRA Trigger Plugin](https://plugins.jenkins.io/jira-trigger/) which included Jersey Core 1.5 (which includes `javax.ws.rs.core..MultivaluedMap` from JAX-RS 1.x). After that plugin was removed, it works.
